staging/src/k8s.io/apiserver/pkg/audit/types.go

```diff
@@ -34,4 +34,8 @@ type Backend interface {
 	// Run will initialize the backend. It must not block, but may run go routines in the background. If
 	// stopCh is closed, it is supposed to stop them. Run will be called before the first call to ProcessEvents.
 	Run(stopCh <-chan struct{}) error
+
+	// Shutdown will synchronously shut down the backend while making sure that all pending
+	// events are delivered. This method blocks until the stopCh passed to the Run method is closed.
```

Comment:
put less requirements on the implementation. I.e. my variant matches an empty `Shutdown() {}` implementation. Your's doesn't.